sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java

```diff
@@ -1015,7 +1017,23 @@ public PDone apply(PCollection<TableRow> input) {
           table.setProjectId(options.getProject());
         }
         String jobIdToken = UUID.randomUUID().toString();
-        String tempFilePrefix = options.getTempLocation() + "/BigQuerySinkTemp/" + jobIdToken;
+        String tempLocation = options.getTempLocation();
```

Comment:
Done, thanks!

Removed the check of gcs path, since it is done in validate(). 
In there (line 993), I am using GcsPath directly and expect it fails for non gcs path.
I am not checking the GCSIOChannelFactory, because I think there might be other implementations.